### PR TITLE
Added UPS-MIB Trap On Battery (upsTraps.0.1)

### PR DESCRIPTION
**Pull request by TheGreatDoc**
Added UPS-MIB::upsTraps.0.1 support.

This allows for instant notification when an UPS goes on battery state.

It updates database with current `Output source` state (on battery), `Time on battery` and `Estimated battery time remaining`

Example:
`UPS running on battery for 120 seconds. Estimated 100 minutes remaining`

DO NOT DELETE THE UNDERLYING TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [X] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)
- [ ] If my Pull Request does some changes/fixes/enhancements in the WebUI, I have inserted a screenshot of it.

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

File: tests/Feature/SnmpTraps/UpsTrapsOnBatteryTest.php [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+        $trapText = "$device->hostname
+UDP: [$device->ip]:161->[192.168.5.5]:162
+DISMAN-EVENT-MIB::sysUpTimeInstance 9:22:15:00.01
+SNMPv2-MIB::snmpTrapOID.0 UPS-MIB::upsTraps.0.1
+UPS-MIB::upsEstimatedMinutesRemaining.0 100 minutes
+UPS-MIB::upsSecondsOnBattery.0 120 seconds
+UPS-MIB::upsConfigLowBattTime.0 1 minutes";
+
+        \Log::shouldReceive('warning')->never()->with("Snmptrap UpsTraps: Could not find matching sensor \'Estimated battery time remaining\' for device: " . $device->hostname);
```
Comment (murrant) [inline tests/Feature/SnmpTraps/UpsTrapsOnBatteryTest.php:53]: why are you using never()?  This actually changes it to shouldNotRecieve()...

Comment (TheGreatDoc) [reply tests/Feature/SnmpTraps/UpsTrapsOnBatteryTest.php:53]: Actually, I took that example from the jnx* trap tests, for example here https://github.com/librenms/librenms/blob/300645388fefdcd999bc6dff4738b08959c6abac/tests/Feature/SnmpTraps/JnxBgpM2Test.php#L61
But I dont mind changin them to shouldNotReceive()

Comment (TheGreatDoc) [reply tests/Feature/SnmpTraps/UpsTrapsOnBatteryTest.php]: @murrant I've changed without looking and shouldNotReceive doesnt exists...

File: LibreNMS/Snmptrap/Handlers/UpsTrapsOnBattery.php [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * @package    LibreNMS
+ * @link       http://librenms.org
+ * @author     TheGreatDoc
```
Comment (murrant) [inline LibreNMS/Snmptrap/Handlers/UpsTrapsOnBattery.php:22]: author should be your real name :D

**Comment by murrant** [comment]
This pull request has been mentioned on **LibreNMS Community**. There might be relevant details there:

https://community.librenms.org/t/v1-65-release-changelog-june-2020/12687/1

**Comment by murrant** [comment]
This pull request has been mentioned on **LibreNMS Community**. There might be relevant details there:

https://community.librenms.org/t/v1-65-release-changelog-june-2020/12687/1

**Pull request merged**